pkgs/by-name/co/cosmic-comp/package.nix

```diff
@@ -55,7 +71,7 @@ rustPlatform.buildRustPackage {
 
   # These libraries are only used by the X11 backend, which will not
   # be the common case, so just make them available, don't link them.
-  postInstall = ''
+  postInstall = lib.optionalString useXWayland ''
     wrapProgram $out/bin/cosmic-comp \
       --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [
         xorg.libX11 xorg.libXcursor xorg.libXi xorg.libXrandr
```

Comment:
These are unrelated to Xwayland — they're for running the compositor itself as an X11 window.  So they shouldn't be behind the same option.

Reply:
fixed